pkg/gc/gc.go

```diff
@@ -0,0 +1,189 @@
+/*
+ *     Copyright 2020 The Dragonfly Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gc
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// GC is the interface used for release resource
+type GC interface {
+	// Add adds GC task
+	Add(string, Task)
```

Comment:
The Task interface also needs to have a timeout and an execution interval

Reply:
There is no such demand now.
https://wiki.c2.com/?PrematureOptimization